src/Utils/Utils.php

```diff
@@ -471,4 +474,49 @@ public static function withErrorHandling(callable $fn, array &$errors)
             }
         };
     }
-}
+    
+    /**
+     * @param $traversable
+     * @param callable $valueFn function($value, $key) => $newValue
+     * @return array
+     * @throws \Exception
+     */
+    public static function mapValues($traversable, callable $valueFn) {
+        self::invariant(is_array($traversable) || $traversable instanceof \Traversable, __METHOD__ . ' expects array or Traversable');
+
+        $map = [];
+        foreach ($traversable as $key => $value) {
+            $newValue = $valueFn($value, $key);
+            $map[$key] = $newValue;
+        }
+        return $map;
+    }
```

Comment:
How is this function different from regular `map()` in the same class, they look identical?

Reply:
I removed mapValues, wasn't aware at the time of all the porting you did there with different names than JS version